doc/specs/#605 - Search/spec.md

```diff
@@ -0,0 +1,91 @@
+5. The search dialog should not block terminal's view. 
+
+Conhost already has a module for search. It realizes case sensitive or insensitive exact text match search, and it provides methods to select the found word. However, we want to make search as a shared component between Terminal and Console host. Now search module is part of Conhost, and its dependencies include BufferOut and some other types in ConHost such as SCREEN_INFORMATION. In order to make Search a shared component, we need to remove its dependency on ConHost types. BufferOut is already a shared component, but we need to make sure there is no other Conhost dependency.
+
+Search process:
+Search is performed on a XAML AutoSuggestBox. Once the user click the "Find" icon, we start from the cursor (or the current selection), and try to find the exact text in the text buffer. The nearest searched one will be selected. And we set the search start point to the selected text. The next time "Find" button is clicked, the search will start before or after the previous searched text.
+The user can choose to search up or down by choosing up arrow or down arrow buttons.
+The user can choose to do case sensitive or insensitive match by clicking a button.
+If the user click the "X" button, the search stopped and the search box disappears. In phrase one we do not store any state, but in the future we can consider storing the search history. 
+
+## UI/UX Design
+
+![Sol Design](images/AutoSuggestBox.png)
+
+Above is the XAML AutoSuggestBox. We also need to add at least two more buttons, one for case sensitive/insensitive switch, and one for search direction switch. We can consider adding the buttons on the right side of the AutoSuggestBox.
+
+The search box should be on the top right corner of the Terminal window. We need to avoid it blocking too much screen contents. 
+
+## Capabilities
+
+1. The user can search exact matched text in the text buffer of the Terminal Screen. 
+2. The user can choose to search case sensitively and insensitively. 
+3. The user can search up or down. 
+4. Found text will be selected. 
+5. The search will start from the active selected text (inclusive) if there is one, or where the cursor is. 
+5. The search will automatically go around when it reaches the starting point. 
+
+### Accessibility
+
+This feature could help multitaksing terminal users. 
+
+### Security
+
+This feature should not introduce any new security issues.
+
+### Reliability
+
+This feature enable users to search for text in the terminal input/output history. This is a widely-used feature in most editors and thus improve the reliability of Terminal once added. 
+
+### Compatibility
+
+This feature won't break existing features of Terminal.
+
+### Performance, Power, and Efficiency
+
+This feature only launches in need. It does not impact the performance of Terminal. 
+
+## Potential Issues

```

Comment:
What happens when the window isn't big enough for the full dialog to be visible?

Reply:
Oh, this is a issue that I should think about.

Reply:
The search box will be "squeezed":

![image](https://user-images.githubusercontent.com/11893272/69596072-3e2fb580-0fb6-11ea-9975-1442a4238c2d.png)

Reply:
Yes, it is similar to VSCode: 

![image](https://user-images.githubusercontent.com/11893272/69668381-a036fc00-1044-11ea-86b7-e62f5b546839.png)

Reply:
If the Terminal is too short, the Terminal screen will disappear though.

Reply:
Added